src-gui/src/renderer/components/pages/swap/swap/init/walletAndMakers/WalletAndMakers.tsx

```diff
@@ -0,0 +1,184 @@
+    )
+    .sortBy((quote) => quote.quoteWithAddress.quote.price)
+    .sortBy((quote) => (quote.requestId ? 0 : 1))
+    .uniqBy((quote) => quote.quoteWithAddress.peer_id)
+    .value();
+
+  return (
+    <>
+      <Box
+        sx={{
+          display: "flex",
+          flexDirection: "column",
+          gap: 3,
+        }}
+      >
+        <Paper
+          elevation={8}
+          sx={{
+            padding: 2,
+            display: "flex",
+            flexDirection: { xs: "column", md: "row" },
+            gap: 2,
+          }}
+        >
+          <Box sx={{ flexGrow: 1, flexShrink: 0, minWidth: "12em" }}>
+            <Typography variant="body1">Bitcoin Balance</Typography>
+            <Typography variant="h5">
+              <SatsAmount amount={max_giveable} />
+            </Typography>
+          </Box>
+
+          <Divider
+            orientation="vertical"
+            flexItem
+            sx={{
+              marginX: { xs: 0, md: 1 },
+              marginY: { xs: 1, md: 0 },
+              display: { xs: "none", md: "block" },
+            }}
+          />
+          <Divider
+            orientation="horizontal"
+            flexItem
+            sx={{
+              marginX: { xs: 0, md: 1 },
+              marginY: { xs: 1, md: 0 },
+              display: { xs: "block", md: "none" },
+            }}
+          />
+
+          <Box
+            sx={{
+              flexShrink: 1,
+              display: "flex",
+              flexDirection: { xs: "row", md: "column", lg: "row" },
+              gap: 2,
+            }}
+          >
+            <Box
+              sx={{
+                display: "flex",
+                flexDirection: "column",
+                gap: 1,
+              }}
+            >
+              <Typography variant="body1">Deposit</Typography>
+              <Typography variant="body2" color="text.secondary">
+                Send Bitcoin to your internal wallet to swap your desired amount
+                of Monero
+              </Typography>
+              <ActionableMonospaceTextBox content={deposit_address} />
+            </Box>
+            <Box
+              sx={{
+                display: "flex",
+                justifyContent: "center",
+                width: "100%",
+                maxWidth: "8em",
+              }}
+            >
+              <BitcoinQrCode address={deposit_address} />
+            </Box>
+          </Box>
+        </Paper>
+
+        {/* Available Makers Section */}
+        <Box>
+          <Box
+            sx={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "space-between",
+              mb: 2,
+            }}
+          >
+            <Typography variant="h5">Select an offer</Typography>
+          </Box>
+
+          {/* Maker Discovery Status */}
+          <MakerDiscoveryStatus />
+
+          {/* Real Maker Offers */}
+          <Box>
+            {pendingSelectMakerApprovals.length > 0 && (
```

Comment:
We need to check the length of `makerOffers` here. If we know of a bunch of quotes but none are acceptable yet, we won't display anything.